diff --git a/CMakeLists.txt b/CMakeLists.txt
index a0a46da..266c089 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -240,3 +240,11 @@ add_custom_target(dist COMMAND ${CMAKE_MAKE_PROGRAM} package_source)
 ################################################################################
 
 add_subdirectory(utils)
+
+install(
+  DIRECTORY aircraft data_output data_plot engine scripts systems
+  DESTINATION etc/JSBSim
+  PATTERN *.cvsignore EXCLUDE
+  PATTERN INSTALL EXCLUDE
+  PATTERN Makefile.am EXCLUDE
+  )
diff --git a/src/FGFDMExec.cpp b/src/FGFDMExec.cpp
index a92cb11..3baf93a 100644
--- a/src/FGFDMExec.cpp
+++ b/src/FGFDMExec.cpp
@@ -200,8 +200,8 @@ FGFDMExec::~FGFDMExec()
   ChildFDMList.clear();
 
   PropertyCatalog.clear();
-  
-  SetGroundCallback(0);
+
+  // SetGroundCallback(0);
 
   if (FDMctr != 0) (*FDMctr)--;
 
